src/lolex-src.js

```diff
@@ -196,6 +208,9 @@ function addTimer(clock, timer) {
     timer.type = timer.immediate ? "Immediate" : "Timeout";
 
     if (timer.hasOwnProperty("delay")) {
+        if (!isNumberFinite(timer.delay)) {
```

Comment:
So I checked - the spec actually says (if < 0 then = 0) but Infinity isn't explicitly treated - >Int32 numbers overflow and also get set to 0 but not explicitly. 

Node on the other hand sets the value to 1 (and not 0) https://github.com/nodejs/node/blob/master/lib/internal/timers.js#L19 done in https://github.com/nodejs/node/blob/master/lib/internal/timers.js#L55

Reply:
https://github.com/nodejs/node/blob/master/lib/internal/timers.js#L55

Reply:
(I've never read much spec, so all questions are in good faith)

From the linked issue in Jest, I read it as point 8 in https://heycam.github.io/webidl/#abstract-opdef-converttoint (as it's a long, not `EnforceRange`), namely

>If x is NaN, +0, +∞, or −∞, then return +0.

Reply:
Yeah, exactly.

Reply:
Oh, so we are in agreement. Cool! 😀 

Do you think it's a bug in Node?

Reply:
@bnoordhois https://github.com/nodejs/node/commit/7fc835afe362ebd30a0dbec81d3360bd24525222 - 

Do you happen to remember why timers are set to 1 and not 0?

Reply:
Seeing as the tag failed, I'll be a bit cheeky and do @bnoordhuis

Reply:
So glad you guys are looking into the sweaty details of this. Delegation is :crown: Hope you add some comments to the implementation to make code archaeology easier for future maintainers :)

Reply:
I've opened an issue over at core about the bahvior - I think it's fine to be honest, it's not what the timers spec says but it's what Chrome and Node do basically https://github.com/nodejs/node/issues/20596

Reply:
FWIW, I'm fine setting it to `1`. That's not observable from my test - the test is the order in which the callbacks gets invoked, not if the timeout is `0` or `1`.